prov/verbs/src/verbs_info.c

```diff
@@ -1584,25 +1584,6 @@ static int fi_ibv_handle_ib_ud_addr(const char *node, const char *service,
 	return ret;
 }
 
-static void fi_ibv_remove_nosrc_info(struct fi_info **info)
-{
-	struct fi_info **fi = info, *next;
-	while (*fi && ((*fi)->ep_attr->type == FI_EP_MSG)) {
```

Comment:
This loop assumes that all fi_info structures associated with msg endpoints are grouped together.  Is that always the case, even when XRC is used?

Reply:
Good point. It seems the MSG endpoints may not be grouped together (based on the loop in fi_ibv_init_info) and this function would miss to remove some "no src fi_info". I think the issue wasn't caught since apps are most likely to use the first matching fi_info.

The newer patch would not have this problem as we iterate through the complete list of devices and endpoint types to remove any fi_info that doesn't have IPoIB address configured.